src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -24,7 +24,12 @@ public LinkedInAuthenticationHandler([NotNull] HttpClient client)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
-            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            var endpoint = string.Empty;
+            if (Options.Fields.Count > 0) {
+                endpoint = $"{Options.UserInformationEndpoint}/~:({string.Join(",", Options.Fields)})";
```

Comment:
> Since the url doesn't use a query string, the QueryHelpers won't help.

People are free to specify custom query string parameters in `Options.UserInformationEndpoint`. The OAuth2 providers developed by the ASP.NET Core team have been fixed to make this scenario possible and I'm a bit concerned about making the LI provider inconsistent: https://github.com/aspnet/Security/issues/365

That said, given the weird syntax used by LI, we'd have to parse the `Options.UserInformationEndpoint` provided by the user to insert the fields "component" at the right place, which doesn't seem rock-solid.

@jerriep thoughts?